torch_dftd/torch_dftd3_calculator.py

```diff
@@ -97,14 +97,22 @@ def _preprocess_atoms(self, atoms: Atoms) -> Dict[str, Optional[Tensor]]:
         )
         Z = torch.tensor(atoms.get_atomic_numbers(), device=self.device)
         if any(atoms.pbc):
-            cell = torch.tensor(
+            cell: Optional[Tensor] = torch.tensor(
                 atoms.get_cell(), device=self.device, dtype=self.dtype, requires_grad=True
             )
         else:
             cell = None
         pbc = torch.tensor(atoms.pbc, device=self.device)
         edge_index, S = self._calc_edge_index(pos, cell, pbc)
-        input_dicts = dict(pos=pos, Z=Z, cell=cell, pbc=pbc, edge_index=edge_index, shift=S)
+        if cell is None:
+            shift = S
+        else:
+            # shift = S
+            shift = torch.mm(S, cell.detach())
```

Comment:
`shift` contains different data from the previous one.
Current `shift` equals to `previous_shift * cell`
This is because `shift` is used for `requires_grad_(True)` directly instead of `cell`.